inst/package_report/package_inheritance_reporter.Rmd

```diff
@@ -0,0 +1,24 @@
+---
+title: "Package Class Inheritance Report"
+output: html_document
+---
+
+## Class Inheritance
+
+Here's an overview of the classes defined in **`r reporter$pkg_name`**.
```

Comment:
Because this is so complicated, I actually think you should duplicate your explanation from roxygen:

* what a "class" is
* what the types are
* how we come up with this

into this report.

Up to you if you want to  do it in this top part or a new section at the bottom.